src/core_plugins/kibana/server/lib/manage_uuid.js

```diff
@@ -0,0 +1,73 @@
+import uuid from 'node-uuid';
+import Promise from 'bluebird';
+import { hostname } from 'os';
+import { join as pathJoin } from 'path';
+import { readFile as readFileCallback, writeFile as writeFileCallback } from 'fs';
+
+const serverHostname = hostname();
+const FILE_ENCODING = 'utf8';
+
+export default async function manageUuid(server) {
+  const config = server.config();
+  const serverPort = server.info.port;
+  const fileName = `${serverHostname}:${serverPort}`;
+  const uuidFile = pathJoin(config.get('path.data'), fileName);
```

Comment:
Maybe we should make the uuid file path configurable still. I agree with @jbudz that it would be simpler if the path was just `join(config.get('path.data'), 'uuid')` and the user was expected to assign multiple instances of Kibana their own `path.data`. This is contrary to the way Elasticsearch works, but the first thing elasticsearch does is split the data directory up by node number... Another option that doesn't bind this path to the port number.